src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs

```diff
@@ -137,6 +139,19 @@ public async Task Start()
 
             logger.LogInformation($"Omnisharp server running using Lsp at location '{environment.TargetDirectory}' on host {environment.HostProcessId}.");
 
+            await Task.WhenAll(
+                _compositionHost
+                    .GetExports<IProjectSystem>()
+                    .Select(ps => ps.WaitForIdleAsync())
+                    .ToArray());
+
+            _workDoneObserver?.OnNext(new WorkDoneProgressReport
+            {
+                Message = "Language Server ready",
+                Percentage = 100,
+            });
+            _workDoneObserver?.OnCompleted();
```

Comment:
I don't think we want to wait for project loading to complete here. You could extract it to a method and just fire and forget.

Reply:
Actually, why not move this back to the Initialized method?

Reply:
Because in order to send lsp-progress messages to client server must be initialized.
Otherwise there are warnings that server it not initialized and messages about initialization are got all at once after server is initilized.